tests/integration_tests/test_mongo_task.py

```diff
@@ -474,11 +495,130 @@ def test_filter_on_data_categories_mongo(
         target_categories,
         dataset_graph.data_category_field_mapping,
     )
+
+    # Test for accessing array
     assert filtered_results["mongo_test:internal_customer_profile"][0] == {
         "derived_interests": ["marketing", "food"]
     }
 
 
+@pytest.mark.integration
+def test_array_querying_mongo(
+    db,
+    privacy_request,
+    example_datasets,
+    policy,
+    integration_mongodb_config,
+    integration_postgres_config,
+):
+
+    postgres_config = copy.copy(integration_postgres_config)
+
+    dataset_postgres = FidesopsDataset(**example_datasets[0])
+    graph = convert_dataset_to_graph(dataset_postgres, integration_postgres_config.key)
+    dataset_mongo = FidesopsDataset(**example_datasets[1])
+    mongo_graph = convert_dataset_to_graph(
+        dataset_mongo, integration_mongodb_config.key
+    )
+    dataset_graph = DatasetGraph(*[graph, mongo_graph])
+
+    access_request_results = graph_task.run_access_request(
+        privacy_request,
+        policy,
+        dataset_graph,
+        [postgres_config, integration_mongodb_config],
+        {"email": "jane@example.com"},
+    )
+
+    target_categories = {"user.derived"}
+    filtered_results = filter_data_categories(
+        access_request_results,
+        target_categories,
+        dataset_graph.data_category_field_mapping,
+    )
+    # Array field mongo_test:internal_customer_profile.customer_identifiers contains identity
+    # Only matching identity returned
+    assert filtered_results["mongo_test:internal_customer_profile"][0][
+        "customer_identifiers"
+    ]["derived_emails"] == ["jane@example.com"]
+
+    # # Entire derived_interests array returned - this is not an identity or "to" reference field
+    assert filtered_results["mongo_test:internal_customer_profile"][0][
+        "derived_interests"
+    ] == ["interior design", "travel", "photography"]
+
+    filtered_identifiable = filter_data_categories(
+        access_request_results,
+        {"user.provided.identifiable"},
+        dataset_graph.data_category_field_mapping,
+    )
+
+    # Includes array field
+    assert filtered_identifiable["mongo_test:customer_details"] == [
+        {
+            "birthday": datetime(1990, 2, 28, 0, 0),
+            "gender": "female",
+            "children": ["Erica Example"],
+        }
+    ]
+
+    # items in array mongo_test:customer_details.travel_identifiers used to lookup matching array elements
+    # in mongo_test:flights:passenger_information.passenger_ids.  passenger_information.full_name has relevant
+    # data category.
+    assert filtered_identifiable["mongo_test:flights"][0] == {
+        "passenger_information": {"full_name": "Jane Customer"}
+    }
```

Comment:
If we're only expecting `filtered_identifiable["mongo_test:flights"]` to have one element, it would be good to assert that here also

Reply:
good idea, I can add that assertion

Reply:
added!